cloudinit/cmd/devel/logs.py

```
@@ -143,15 +144,31 @@ def _copytree_rundir_ignore_files(curdir, files):
 
 def _write_command_output_to_file(cmd, filename, msg, verbosity):
     """Helper which runs a command and writes output or error to filename."""
+    output = ""
     try:
-        out, _ = subp(cmd)
+        ensure_dir(os.path.dirname(filename))
+        output = subp(cmd)[0]
+        write_file(filename, output)
+    except ProcessExecutionError as e:
+        write_file(filename, str(e))
+        _debug("collecting %s failed.\n" % msg, 1, verbosity)
+        return str(e)
```

Do we want to return the exception text here? How would it be used?